### PR TITLE
Production tests for fmcomms 2 and 3

### PR DESCRIPTION
**Pull request by ranechita**
# Description

Production tests for fmcomms2/3, similar to the test profiles from iio-oscilloscope.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How has this been tested?

**Test Configuration**:
* Hardware: ADRV9361-ZC702
* OS: Ubuntu 20.04

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/test_fmcomms2-3_p.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    ],
+)
+def test_ad9361_iq_loopback(test_iq_loopback, iio_uri, classname, channel, param_set):
+    test_iq_loopback(iio_uri, classname, channel, param_set)
+
+
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+def test_dcxo(test_dcxo_calibration, classname, iio_uri):
+    test_dcxo_calibration(classname, iio_uri)
```
Comment (tfcollins) [inline test/test_fmcomms2-3_p.py:191]: Can we implement a pytest.skip somewhere so if the instrument is missing users get notified? pytest will hide prints if not using the -s flag.

Comment (ranechita) [reply test/test_fmcomms2-3_p.py:191]: Thank you for the observation, I replaced the print with pytest.skip in dcxo_calibration and it works just fine when i turn off the device

File: test/scpi.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    full_uri = iio_uri.split(":", 2)
+    if full_uri[0] != "ip":
+        print("Tuning currently supported only for ip URIs")
+    ip = full_uri[1]
+
+    ssh_client = paramiko.SSHClient()
+    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh_client.connect(ip, username="root", password="analog")
+    ssh_stdin, ssh_stdout, ssh_stderr = ssh_client.exec_command(
+        "cat /sys/kernel/debug/clk/ad9361_ext_refclk/clk_rate"
```
Comment (tfcollins) [inline test/scpi.py:82]: Can we use IIO to get this value? If not can we add a check to the SSH stderr to make sure we know if that breaks

Comment (ranechita) [reply test/scpi.py:83]: I added a check, is simply printing the stderr in order?

Comment (tfcollins) [reply test/scpi.py:83]: Is there an error code we can check here? Like if not 0 raise Exception

**Pull request merged**